packages/components/src/components/hds/form/key-value-inputs/index.hbs

```diff
@@ -0,0 +1,116 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+<fieldset
+  class="hds-form-key-value-inputs"
+  ...attributes
+  {{style --hds-key-value-pair-columns=this._gridTemplateColumns}}
+  {{! need to set aria-labelledby because the legend is not a direct descendant of the fieldset }}
+  aria-labelledby="{{this.id}}-legend"
+  {{! @glint-expect-error }}
+  aria-describedby={{this.ariaDescribedBy}}
+  tabindex="-1"
+  {{this._setUpKeyValueInputs}}
+>
+  <div class="hds-form-key-value-inputs__header">
+    {{yield
+      (hash
+        Legend=(component
+          "hds/form/legend"
+          contextualClass="hds-form-key-value-inputs__legend"
+          id=(concat this.id "-legend")
+          isOptional=@isOptional
+          isRequired=@isRequired
+        )
+      )
+      to="header"
+    }}
+    {{yield
+      (hash
+        HelperText=(component
+          "hds/form/helper-text"
+          contextualClass="hds-form-key-value-inputs__helper-text"
+          controlId=this.id
+          onInsert=this.appendDescriptor
+        )
+      )
+      to="header"
+    }}
+  </div>
+
+  <div>
+    {{yield (hash Generic=(component "hds/yield")) to="header"}}
+  </div>
```

Comment:
[issue] I looked at the commits history and I see you've wrapped it in a `<div>` element, why? it's adding an extra empty `div` even if there's no generic yielded content, and this causes extra space (because of the `gap` between flex items)
<img width="452" alt="screenshot_5129" src="https://github.com/user-attachments/assets/f1827ca7-2310-457f-bc8b-572e8487a8f5" />

Reply:
(one consideration is that the generic yield could live under the `<div class="hds-form-key-value-inputs__header">` and delegate responsibility of the spacing before it to the consumers (like we do in other components, eg. the Alert)

Reply:
and potentially have the `Legend`, `HelperText` and `Generic` all under the same `{{yield ...` declaration (we do this in the Form Layout components, for example, leaving the responsibility to the consumers to use the yielded elements in the correct order (which is what we do in the `Footer` below, anyway)

Reply:
Ah, I did it because the generic content in the header is supposed to have a 12px gap between it and the rest of the header content. Since the whole fieldset is a flexbox with a 12px rowgap, I thought moving it out of the main header yield it would be an easy solution to achieve the spacing.

Wrapping it in a new div tho is entirely unnecessary, I can remove that.